the-collab-lab/tcl-71-smart-shopping-list
# Dialogs

## Description

This PR normalizes adds accessible dialogs using the native HTML5 <dialog> element to prompt confirmation from the user when deleting a list or an item.

It also includes some minor bug-fixes, such as the spinner background color, the Layout/main element height, and the hover behavior of the buttons on the List and ManageList views.

## Related Issue

closes #69

## Acceptance Criteria

- [x] When prompting the user for confirmation, a dialog should be used.
- [x] The dialog should adhere to accessibility guidelines, and be fully controllable via keyboard.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :sparkles: New feature     |
|  ✓    | :art:  Enhancement |
|  ✓    | :bug:  Bug fix |

## Updates

### Before

![image](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/86715948/5a6fdd92-8b1d-450f-9531-8d2f43ea76a1)

### After

![image](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/86715948/0c8fb259-2581-4a81-92de-b488ffae7e40)

![image](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/86715948/fc546a32-207d-4bf4-b629-acbfe4d3af95)

![image](https://github.com/the-collab-lab/tcl-71-smart-shopping-list/assets/86715948/7add3cb1-08f7-4494-a5c8-acb448b70805)

## Testing Steps / QA Criteria

- `git pull`
- `git checkout dialogs`
- Check behavior when deleting lists and items.
